abstract_commands.xml

```diff
@@ -2,13 +2,13 @@
     <register name="Access Register">
         \begin{steps}{Perform the following sequence of operations:}
         \item Halt the hart, if \Fprehalt is set. If the hart is already
-            handled, the entire command fails.
+            halted, the entire command fails.
         \item Execute the Program Buffer, if \Fpreexec is set.
         \item Copy data from the register specified by \Fregno into {\tt data}, if \Fwrite is clear.
         \item Copy data from {\tt data} into the register specified by \Fregno, if \Fwrite is set.
         \item Execute the Program Buffer, if \Fpostexec is set.
-        \item Resume the hart, if \Fpostresume is set. If the hart is already
-            running, the entire command fails.
+        \item Resume the hart, if \Fpostresume is set. If the hart is already 
+        running and \Fprehalt is not set, the entire command fails.
```

Comment:
Yes, I had imagined checking at the time the resume would happen. Eg. what do you do if the debugger writes a dret to the program buffer, or something else weird happens? Checking at the time of execution feels the most robust to me, and not complicated.

Checking beforehand is nicer in theory, but there may be more corner cases that you miss.